common/src/auth/ServiceConnectionString.cs

```diff
@@ -0,0 +1,156 @@
+        /// <exception cref="ArgumentNullException">if the provided parser is null.</exception>
+        public ServiceConnectionString(ServiceConnectionStringParser parser)
+        {
+            if (parser == null)
+            {
+                throw new ArgumentNullException(nameof(parser));
+            }
+
+            HostName = parser.HostName;
+            SharedAccessKeyName = parser.SharedAccessKeyName;
+            SharedAccessKey = parser.SharedAccessKey;
+            SharedAccessSignature = parser.SharedAccessSignatureString;
+            ServiceName = parser.ServiceName;
+            HttpsEndpoint = new UriBuilder("https", parser.HostName).Uri;
+        }
+
+        /// <summary>
+        /// The Service Name
+        /// </summary>
+        public string ServiceName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Client Hostname
+        /// </summary>
+        public string HostName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Client Https Endpoint
+        /// </summary>
+        public Uri HttpsEndpoint
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Audience
+        /// </summary>
+        public string Audience
+        {
+            get { return HostName; }
+        }
+
+        /// <summary>
+        /// The Service Access Policy Name
+        /// </summary>
+        public string SharedAccessKeyName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Shared Access Key for the specified
+        /// access policy
+        /// </summary>
+        public string SharedAccessKey
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Shared Access Signature
+        /// </summary>
+        public string SharedAccessSignature
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Returns the shared access signature for authorization
+        /// </summary>
+        /// <returns></returns>
+        public string GetSasToken()
+        {
+            string password;
+            if (string.IsNullOrWhiteSpace(SharedAccessSignature))
+            {
+                TimeSpan timeToLive;
+                password = BuildToken(out timeToLive);
+            }
+            else
+            {
+                password = SharedAccessSignature;
+            }
+
+            return password;
+        }
+
+        /// <summary>
+        /// Parser for the Service Connection String
+        /// </summary>
+        /// <param name="connectionString"> The DPS Connection String </param>
+        /// <returns></returns>
+        public static ServiceConnectionString Parse(string connectionString)
```

Comment:
This is a factory, akin to a ctor. Any reason it doesn't belong up by the ctor?

Reply:
done